packages/analyzer/src/model/class.ts

```diff
@@ -211,6 +216,7 @@ export class Class implements Feature, DeclaredWithStatement {
     if (init.methods !== undefined) {
       this._overwriteInherited(this.methods, init.methods, undefined, true);
     }
+    this.constructorMethod = init.constructorMethod;
```

Comment:
We need to apply some logic similar to `_overwriteInherited` in `inheritFrom` on the constructor, to be consistent with properties and regular methods.

Reply:
Had a stab at it. Is this what you were thinking?